src/crypto/ring.rs

```diff
@@ -11,14 +11,20 @@ impl From<ring::error::Unspecified> for CryptoError {
     }
 }
 
+impl From<std::convert::Infallible> for CryptoError {
```

Comment:
For completeness, why did we add this?

Reply:
The most accurate answer is "the compiler told me to".  I tried looking through the ring commits to see what breaking changes had occurred, and 50 commits in I had no idea.

But more precisely, one of those `try_from()?`'s returns `Result<, Infallible>`, and this impl made the types check out in that case.  If there's a better approach I'm all :eyes: :)